exporters/common/src/main/java/io/opentelemetry/exporter/internal/TlsConfigHelper.java

```diff
@@ -5,79 +5,76 @@
 
 package io.opentelemetry.exporter.internal;
 
-import com.google.common.annotations.VisibleForTesting;
-import java.util.logging.Logger;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
 import javax.annotation.Nullable;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
 import javax.net.ssl.SSLException;
-import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
 import javax.net.ssl.X509KeyManager;
 import javax.net.ssl.X509TrustManager;
 
 /**
- * Utility class to help with management of TLS related components. This class is ultimately
- * responsible for enabling TLS via callbacks passed to the configure[...]() methods. This class is
- * only intended for internal OpenTelemetry exporter usage and should not be used by end-users.
+ * Utility class to help with management of TLS related components. TLS config consists {@link
+ * #keyManager}, {@link #trustManager}, which combine to form {@link #sslContext}. These components
+ * can be configured via higher level APIs ({@link #createTrustManager(byte[])} and {@link
+ * #createKeyManager(byte[], byte[])}) which parse keys in PEM format, or the lower level API {@link
+ * #setSslContext(SSLContext, X509TrustManager)} in which the components are directly set, but NOT
+ * both. Attempts to reconfigure components which have already been configured throw {@link
+ * IllegalStateException}. Consumers access components via any combination of {@link
+ * #getKeyManager()}, {@link #getTrustManager()}, and {@link #getSslContext()}.
  *
  * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
  * at any time.
  */
 public class TlsConfigHelper {
 
-  private static final Logger logger = Logger.getLogger(TlsConfigHelper.class.getName());
-
-  private final TlsUtility tlsUtil;
-
   @Nullable private X509KeyManager keyManager;
   @Nullable private X509TrustManager trustManager;
-  @Nullable private SSLSocketFactory sslSocketFactory;
+  @Nullable private SSLContext sslContext;
 
-  public TlsConfigHelper() {
-    this(new TlsUtility() {});
-  }
-
-  @VisibleForTesting
-  TlsConfigHelper(TlsUtility tlsUtil) {
-    this.tlsUtil = tlsUtil;
-  }
-
-  /** Sets the X509TrustManager. */
-  public TlsConfigHelper setTrustManager(X509TrustManager trustManager) {
-    this.trustManager = trustManager;
-    return this;
-  }
+  public TlsConfigHelper() {}
 
   /**
-   * Creates a new X509TrustManager from the given cert content.
+   * Configure the {@link X509TrustManager} from the given cert content.
+   *
+   * <p>Must not be called multiple times, or if {@link #setSslContext(SSLContext,
+   * X509TrustManager)} has been previously called.
    *
    * @param trustedCertsPem Certificate in PEM format.
-   * @return this
    */
-  public TlsConfigHelper createTrustManager(byte[] trustedCertsPem) {
+  public void createTrustManager(byte[] trustedCertsPem) {
```

Comment:
If you do decide to keep the functionality as methods instead of constructors, it seems odd to call these methods `create*` when the method returns void.

Reply:
I see where you're coming from, but it's pretty clear from the implementation that it does actually create the trust manager. What about `setTrustManagerFromCert`?

Reply:
`setTrustManagerFromCert` - that's pretty clear.